ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/services/CommandSubmissionServiceAuthorization.scala

```diff
@@ -24,14 +26,22 @@ private[daml] final class CommandSubmissionServiceAuthorization(
 
   override def submit(request: SubmitRequest): Future[Empty] = {
     val effectiveSubmitters = CommandsValidator.effectiveSubmitters(request.commands)
-    authorizer.requireActAndReadClaimsForParties(
+    authorizer.requireActAndReadClaimsForPartiesWithDefaulting(
       actAs = effectiveSubmitters.actAs,
       readAs = effectiveSubmitters.readAs,
       applicationId = request.commands.map(_.applicationId),
+      setApplicationId = (appId, req: SubmitRequest) =>
+        req.update(
+          _.commands.modify(cmd => cmd.update(_.applicationId := appId))
+        ),
+      // req.withCommands(request.commands.map(_.withApplicationId(appId))),
       call = service.submit,
     )(request)
   }
 
+  val testCommandsLens: Lens[SubmitRequest, Commands] = { import SubmitRequest._; _.commands }
+  val testApplicationIdLens: Lens[SubmitRequest, String] = // FIXME: how to build this lens?
```

Comment:
@gerolf-da : preferably I'd hand-over a `Lens[SubmitRequest, ApplicationId]` to the call that checks and defaults the application-id. However, I don't know how to build such a lens with ScalaPB. Can you lend me a hand?